component/config.go

```diff
@@ -164,3 +164,51 @@ var (
 	// DataTypeLogs is the data type tag for logs.
 	DataTypeLogs = mustNewDataType("logs")
 )
+
+// Name is the component name as it is used in the config.
+type Name struct {
+	value string
+}
+
+// String returns the string representation of the name.
+func (n Name) String() string {
+	return n.value
+}
+
+// MarshalText marshals the Name value.
+func (n Name) MarshalText() ([]byte, error) {
+	return []byte(n.value), nil
+}
+
+// nameRegexp is used to validate the name of a component.
+// A name must start with an ASCII alphanumeric character and
+// can only contain ASCII alphanumeric characters, '_', and '-'.
+var nameRegexp = regexp.MustCompile(`^[0-9a-zA-Z][0-9a-zA-Z_-]{0,62}$`)
```

Comment:
What do you think about opening this up to all Unicode letters and numbers? i.e. swapping `[0-9]` for `\pN` (numbers) and `[a-zA-Z]` for `\pL` (letters). Perhaps also replacing the hyphen and dash with `\pP` (punctuation).

I think ASCII is reasonable for type, since there's a specific set of them. On the other hand, the name component is arbitrary and user-defined. So, for example, allowing Chinese characters may be more inclusive of native Chinese speakers/readers/writers.